tests/osaka/eip7594_peerdas/helper_functions.py

```diff
@@ -0,0 +1,240 @@
+"""
+abstract: Tests [EIP-7594: PeerDAS - Peer Data Availability Sampling](https://eips.ethereum.org/EIPS/eip-7594)
+    Tests [EIP-7594: PeerDAS - Peer Data Availability Sampling](https://eips.ethereum.org/EIPS/eip-7594).
+"""  # noqa: E501
+import base64 as b64
+import json
+import math
+
+import ckzg
+
+from spec import Spec, ref_spec_7594
+
+REFERENCE_SPEC_GIT_PATH = ref_spec_7594.git_path
+REFERENCE_SPEC_VERSION = ref_spec_7594.version
+
+def bytes_from_hex(hex_string):
+    """Convert a hex string to bytes."""
+    return bytes.fromhex(hex_string.replace("0x", ""))
+
+def generate_blob_from_hex_byte(hex_byte: str) -> bytes | None:
+    """Take a singular hex byte string and return a valid blob built from this byte."""
+    # preparation (remove 0x if present)
+    if "0x" in hex_byte:
+        hex_byte = hex_byte.replace("0x", "", 1)
+
+    # validity checks
+    if len(hex_byte) != 2:
+        print("You should have passed a singular hex byte like e.g. fe to this function")
+        return None
+
+    if int(hex_byte, 16) > 115:
+        # TODO: figure out why this happens
+        print("For some reason bytes larger than this will lead to errors later in compute_cells(), so it's not allowed for now.")  # noqa: E501
+        return None
+
+    hex_byte = hex_byte.lower()
```

Comment:
Yes, this is because of [`BLS_MODULUS`](https://github.com/ethereum/consensus-specs/blob/cc6996c22692d70e41b7a453d925172ee4b719ad/specs/deneb/polynomial-commitments.md?plain=1#L82). The uint256 value of each field element must be less than this. Given that Python supports large integers, I think it would be best if this function took a seed value, then generated `FIELD_ELEMENTS_PER_BLOB` random field elements and joined them together to make the blob. Something like:

```python
import random

seed = 42
random.seed(seed)
BLS_MODULUS = 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001
ints = [random.randrange(BLS_MODULUS) for _ in range(FIELD_ELEMENTS_PER_BLOB)]
encoded = [i.to_bytes(BYTES_PER_FIELD_ELEMENT, "big") for i in ints]
blob = b"".join(encoded)
```

Reply:
Agreed, great suggestion, and the `seed` can be an int to replace the `hex_byte: str` in the original function.